devices.js

```diff
@@ -370,6 +370,24 @@ const livolo = {
     },
 };
 
+const kwikset = {
+    readUserPinCode: async (endpoint, userId) => {
```

Comment:
I think this method can be inlined, since it's only used once (?).

Also instead of issuing the command directly here, call the `convertGet` of the `pincode_lock` converter to keep code in one place.

Reply:
I tried the second one, and I'm stuck at this (which I assumed I'd hit, which is why I didn't try it the first time around).
```
zigbee2mqtt:info  2020-08-04 20:18:42: MQTT publish: topic 'zigbee2mqtt/Kwikset_Lock_1/action', payload 'pin_code_deleted'
(node:17) UnhandledPromiseRejectionWarning: TypeError: entity.command is not a function
    at Object.getPinCode (/app/node_modules/zigbee-herdsman-converters/converters/toZigbee.js:1489:26)
    at Object.readPinCodeAfterProgramming [as onEvent] (/app/node_modules/zigbee-herdsman-converters/devices.js:383:35)
    at OnEvent.callOnEvent (/app/lib/extension/onEvent.js:32:39)
    at OnEvent.onZigbeeEvent (/app/lib/extension/onEvent.js:17:18)
    at Controller.callExtensionMethod (/app/lib/controller.js:360:44)
(node:17) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 4)
z```

It appears we have a `device` not an `entity` on the onEvent, and I'm not sure specifically how to get to the entity.

Reply:
An entity in the context of zigbee-herdsman-converters is either a group or an endpoint (an object where you can send commands to). I assume this lock defines one endpoint implementing the lock clusters, you should be able to get the endpoint with `device.endpoints[0]`.

Reply:
Thanks, that seems to work.